plugins/Terminal/js/components/commandinput.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import { Map } from 'immutable'
+
+//These commands need a password prompt.
+const specialCommands = [ ['wallet', 'unlock'], ['wallet', 'load', 'seed'], ['help'], ['?'] ]
+
+const CommandInput = ({currentCommand, showCommandOverview, actions}) => {
+	componentDidUpdate: {
+		setTimeout(() => document.getElementById('command-input').focus(), 0)
+	}
+
+	reduce: {
+		const handleTextInput = (e) => {
+			actions.setCurrentCommand(e.target.value)
+		}
+
+		const handleKeyboardPress = (e) => {
+
+			//Enter button.
+			if (e.keyCode === 13) {
+
+				//Check if command is special.
+				switch (  isCommandSpecial(currentCommand, specialCommands) ) {
+				case -1: //Regular command.
```

Comment:
Components should be concerned with presentation and basic functionality.  I'm fine with having a `handleKeyPress` event handler inside a component, but the command spawning logic should be somewhere else.  It's an ideal use case for a saga, ie

```
function *runCommandSaga(action) {
    while (true) {
        const command = yield take(constants.RUN_COMMAND)
        const result = yield spawnCommand(command)
        yield put(receiveResult(result))
    }
}
```